### PR TITLE
nixpkgs-basic-release-checks: recurse into package sets

### PR DESCRIPTION
**Pull request by ncfavier**
This PR does the following:

- use `pkgs/top-level/packages-config.nix` in `nixpkgs-basic-release-checks` in order to check evaluation of packages in package sets like `haskellPackages`. The point of this (apart from just catching more eval failures) is that `packages-config.nix` is also used in `nixos-search` (https://search.nixos.org) to list packages, and if there are any evaluation errors then the import would fail. This hasn't been a problem since now because we haven't actually needed to evaluate packages, but this will change in the future when we also [index package outputs](https://github.com/NixOS/nixos-search/pull/419) since that requires evaluation. Also see https://github.com/NixOS/nix/pull/5922#issuecomment-1021211734 for more context
- consequently, fix evaluation errors in those package sets. This was easier than I thought and was a matter of removing a few references to aliases in `haskellPackages.hs-mesos` and in `rPackages`
- back in `nixpkgs-basic-release-checks`, fix an issue introduced 2 days ago by https://github.com/NixOS/nixpkgs/pull/156587 where the warnings will not be printed if the first `nix-env` errors. Also do some cleaning up by moving common flags into an array and removing `--system-filter \*` which has been the default [for a while](https://github.com/NixOS/nix/blame/1fe3bfdeaf7c57e9f7044a22edfbdf6fb9629e9d/src/nix-env/nix-env.cc#L1375).

Note that an obvious consequence is that the checks should take a bit longer (and more memory) to run. @edolstra do you remember why the variant with `nix-env --meta --xml` was/is needed (introduced, or rather not removed, in https://github.com/NixOS/nixpkgs/commit/1ba5ef6b6a052dd83d4b1666f424b134810a42e3)? Removing it would save some time.

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-nix.nix [hunk 1]
```diff
@@ -140,7 +140,7 @@ self: super: builtins.intersectAttrs super {
 
   hs-mesos = overrideCabal (drv: {
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-nix.nix:141]: Just delete the entire override and add `hs-mesos` to `pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml`, please.

Comment (ncfavier) [reply pkgs/development/haskell-modules/configuration-nix.nix]: Done

Comment (ncfavier) [reply pkgs/development/haskell-modules/configuration-nix.nix]: Do I need to regenerate the package set?

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-nix.nix]: Up to you, doesn't really matter; it'd be done eventually after a couple of days.

**Comment by ncfavier** [comment]
It looks like the basic checks will now take over an hour, as opposed to ~15 minutes currently. Maybe that's not acceptable? I wonder if this should be done in OfBorg instead (cc @grahamc)

**Pull request closed without merging**